### PR TITLE
Add link to user profile

### LINKED ISSUE #13: feature: Add link to user profile
In the markdown output, make the username a link to the github users profile

### PR DESCRIPTION
**Pull request by jellllly420**
# Pull Request

Closes #13.

## Proposed Changes

Add link to user profile.

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

**Comment by jellllly420** [comment]
Looks like @mentions in Github do not render in files like `README.md`. (github/markup#209)

This PR keeps raw @mentions in examples in `README.md` though.

**Review by AlexanderReaper7 — APPROVED** [review]

File: env.py [hunk 1]
```diff
@@ -67,6 +68,15 @@ def get_env_vars() -> (
...
             )
 
+    ltp = os.getenv("LINK_TO_PROFILE")
+    # make sure that ltp is a boolean
+    if ltp:
+        ltp = ltp.lower().strip()
+        if ltp not in ["true", "false", ""]:
+            raise ValueError(
+                "LINK_TO_PROFILE environment variable not a boolean. ie. True or False or blank"
+            )
```
Comment (zkoppert) [inline env.py:78]:
```suggestion
    link_to_profile = os.getenv("LINK_TO_PROFILE")
    # make sure that link_to_profile is a boolean
    if link_to_profile:
        link_to_profile = link_to_profile.lower().strip()
        if link_to_profile not in ["true", "false", ""]:
            raise ValueError(
                "LINK_TO_PROFILE environment variable not a boolean. ie. True or False or blank"
            )
```

File: env.py [hunk 2]
```diff
@@ -82,4 +92,5 @@ def get_env_vars() -> (
         start_date,
         end_date,
         sponsor_info,
+        ltp,
```
Comment (zkoppert) [inline env.py:95]:
```suggestion
        link_to_profile,
```

File: markdown.py [hunk 3]
```diff
@@ -10,6 +10,7 @@ def write_to_markdown(
     organization,
     repository,
     sponsor_info,
+    ltp
```
Comment (zkoppert) [inline markdown.py:13]:
```suggestion
    link_to_profile,
```

File: markdown.py [hunk 4]
```diff
@@ -24,14 +25,15 @@ def write_to_markdown(
         organization (str): The organization for which the contributors are being listed.
         repository (str): The repository for which the contributors are being listed.
         sponsor_info (str): True if the user wants the sponsor_url shown in the report
+        ltp (str): True if the user wants the username linked to Github profile in the report
```
Comment (zkoppert) [inline markdown.py:28]:
```suggestion
        link_to_profile (str): True if the user wants the username linked to Github profile in the report
```

File: markdown.py [hunk 5]
```diff
@@ -24,14 +25,15 @@ def write_to_markdown(
...
 
     Returns:
         None
 
     """
     # Put together the contributor table
     table, total_contributions = get_contributor_table(
-        collaborators, start_date, end_date, organization, repository, sponsor_info
+        collaborators, start_date, end_date, organization, repository, sponsor_info, ltp
```
Comment (zkoppert) [inline markdown.py:36]:
```suggestion
        collaborators, start_date, end_date, organization, repository, sponsor_info, link_to_profile
```

File: markdown.py [hunk 6]
```diff
@@ -127,7 +129,7 @@ def get_summary_table(collaborators, start_date, end_date, total_contributions):
 
 
 def get_contributor_table(
-    collaborators, start_date, end_date, organization, repository, sponsor_info
+    collaborators, start_date, end_date, organization, repository, sponsor_info, ltp
```
Comment (zkoppert) [inline markdown.py:132]:
```suggestion
    collaborators, start_date, end_date, organization, repository, sponsor_info, link_to_profile
```

File: markdown.py [hunk 7]
```diff
@@ -140,6 +142,7 @@ def get_contributor_table(
         organization (str): The organization for which the contributors are being listed.
         repository (str): The repository for which the contributors are being listed.
         sponsor_info (str): True if the user wants the sponsor_url shown in the report
+        ltp (str): True if the user wants the username linked to Github profile in the report
```
Comment (zkoppert) [inline markdown.py:145]:
```suggestion
        link_to_profile (str): True if the user wants the username linked to Github profile in the report
```

File: markdown.py [hunk 8]
```diff
@@ -177,7 +180,7 @@ def get_contributor_table(
                 commit_urls += url + ", "
         new_contributor = collaborator.new_contributor
 
-        row = f"| {username} | {contribution_count} |"
+        row = f"| {'' if ltp == 'false' else '@'}{username} | {contribution_count} |"
```
Comment (zkoppert) [inline markdown.py:183]:
```suggestion
        row = f"| {'' if link_to_profile == 'false' else '@'}{username} | {contribution_count} |"
```

File: test_env.py [hunk 9]
```diff
@@ -34,6 +35,7 @@ def test_get_env_vars(self, mock_getenv):
             start_date,
             end_date,
             sponsor_info,
+            ltp
```
Comment (zkoppert) [inline test_env.py:38]:
```suggestion
            link_to_profile,
```

File: test_env.py [hunk 10]
```diff
@@ -43,14 +45,15 @@ def test_get_env_vars(self, mock_getenv):
         self.assertEqual(start_date, "2022-01-01")
         self.assertEqual(end_date, "2022-12-31")
         self.assertEqual(sponsor_info, "false")
+        self.assertEqual(ltp, 'true')
```
Comment (zkoppert) [inline test_env.py:48]:
```suggestion
        self.assertEqual(link_to_profile, 'true')
```

**Pull request merged**